src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/RangeAttribute.cs

```diff
@@ -84,7 +84,7 @@ public RangeAttribute(Type type, string minimum, string maximum)
         /// parameters, for which the invariant culture is always used for any conversions of the validated value.</remarks>
         public bool ConvertValueInInvariantCulture { get; set; }
 
-        private Func<object, object> Conversion { get; set; }
+        private Func<object, object>? Conversion { get; set; }
```

Comment:
should this be `Func<object?, object?>`?

Reply:
The conversion function isn't user-provided - it's generated internally in SetupConversion based on user-provided minimum and maximum values, and except for the int/double special cases (where there's no null anyway), it will NRE on null inputs (it calls GetType in inputs to see if it needs to actual perform conversions). It's "protected" against this in [IsValid](https://github.com/dotnet/runtime/blob/master/src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/RangeAttribute.cs#L113), where `value` is checked for null explicitly, and invocation is skipped entirely - so the implementation never seems to pass null to the function. So it seems right for the conversion parameter to be non-nullable.

However, it's technically possible (though probably weird) for the user-provided conversion function to *return* null, am changing that.

Re NotNullWhenNotNull, I don't see anything explicit showing a relationship between the nullability of the input and the output of the conversion function - the user could conceivably decide to convert empty strings to null, for example, for the purpose of this attribute.

Let me know what you think of the above.